### PR TITLE
docs: api/grn_table: Move grn_table_sort() reference to the header file

### PR DESCRIPTION
**Pull request by abetomo**
GitHub: GH-1817

Prepare to switch to documents automatically generated by Doxygen.

### CODE REVIEW

File: include/groonga/table.h [hunk 1]
```diff
@@ -505,6 +505,26 @@ struct _grn_table_sort_key {
   int offset;
 };
 
+/**
+ * \brief Sorts the records in table and stores the results in `result`
```
Comment (otegami) [inline include/groonga/table.h:509]:
```suggestion
 * \brief Sort the records in table and store the results in `result`
```

**Review by otegami — APPROVED** [review]
LGTM 👍🏾

File: include/groonga/table.h [hunk 2]
```diff
@@ -505,6 +505,26 @@ struct _grn_table_sort_key {
...
+ * \param table Target table
+ * \param offset Starting offset of the sorted record.
+ *               The records are stored in order from the offset (zero-based).
+ * \param limit Maximum number of records to be stored in `result`
+ * \param result Table to store results
+ * \param keys Array of sort keys
+ *             * keys.key: You can specify either column, accessor, or proc of
+ *               table
+ *             * keys.flags: You can specify the next
+ *               * GRN_TABLE_SORT_ASC: Sort in ascending order
```
Comment (otegami) [inline include/groonga/table.h:521]: How about using `\ref` here like the following PR?
- https://github.com/groonga/groonga/pull/1868
```suggestion
 *               * \ref GRN_TABLE_SORT_ASC
```

**Review by otegami — APPROVED** [review]
LGTM 👍🏾

File: include/groonga/table.h [hunk 3]
```diff
@@ -505,6 +511,26 @@ struct _grn_table_sort_key {
...
+ *
+ * \param ctx The context object
+ * \param table Target table
+ * \param offset Starting offset of the sorted record.
+ *               The records are stored in order from the offset (zero-based).
+ * \param limit Maximum number of records to be stored in `result`
+ * \param result Table to store results
+ * \param keys Array of sort keys
+ *             * keys.key: You can specify either column, accessor, or proc of
+ *               table
```
Comment (kou) [inline include/groonga/table.h:525]:
```suggestion
 *             * keys.key: You can specify either column of table, accessor of table, or proc
```

**Pull request merged**